File: numeropeli.js
#!/usr/bin/env node
var readline = require('readline');

var count_of_questions = 3;
var random_number = Math.floor(Math.random() * (10 - 1) + 1);

console.log("Arvaa numero 1-10 välillä, sinulla on käytettävissä 3 arvausta.");

var rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function ask_number() {
    rl.question("Anna numero >", function(answer) {
        loop(answer);        
    });
}

function loop(answer) {

    count_of_questions -= 1;

    if( count_of_questions <= 0 ) {
        console.log("Käytit kaikki arvauksesi, moi.")
        rl.close();
        return;
    }

    if( answer == random_number ) {
        console.log("Arvasit oikein");
        rl.close();
        return;
    }

    if( answer > random_number ) {
        console.log("Numero on pienempi");
    }        

    if( answer < random_number ) {
        console.log("Numero on suurempi");
    }

    ask_number();        
}

ask_number();

File: numeropeli.py
import random

def ask_number():
    try:
        return int(input("Anna numero >"),10)
    except ValueError:
        return ask_number()

def main():

    amount_of_ask = 3
    rand_number = random.randint(1, 10)

    print("Arvaa numero 1-10 valilla, sinulla on kaytettavissa %s arvausta" % amount_of_ask)

    while amount_of_ask > 0:
        
        user_input = ask_number()   

        if user_input is rand_number:       
            print("Arvasit numeron okein, onnittelu :)")
            break

        if user_input < rand_number:
            print("Numero on suurempi")

        if user_input > rand_number:
            print("Numero on pienempi")

        amount_of_ask -= 1
    else:
        print("Et arvannut numeroa")

if __name__ == '__main__':
    main()